.github/workflows/acquia-backup-databases.yml

```
@@ -0,0 +1,99 @@
+        default: '8.2'
+    secrets:
+      ENVIRONMENT_ID:
+        description: 'Acquia environment ID.'
+        required: true
+      ACLI_CLIENT_ID:
+        description: 'The client ID for authenticating with ACLI.'
+        required: true
+      ACLI_CLIENT_SECRET:
+        description: 'The client secret for authenticating with ACLI.'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  backup-databases:
+    name: 'Backup Acquia databases'
+    runs-on: ubuntu-latest
+    outputs:
+      databases: ${{ steps.get_databases.outputs.database_list }}
+      db_list: ${{ steps.get_databases.outputs.db_list }}
+    steps:
+      - name: Checking out code
+        uses: actions/checkout@v3
+
+      - name: Installing PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ inputs.php_version }}
+
+      - name: Installing dependencies
+        run: |
+          curl -OL https://github.com/acquia/cli/releases/download/2.13.0/acli.phar
+          echo "ACLI downloaded!"
+
+      - name: Authorizing with ACLI
+        run: |
+          php acli.phar auth:login -n \
+            --key="${{ secrets.ACLI_CLIENT_ID }}" \
+            --secret="${{ secrets.ACLI_CLIENT_SECRET }}"
+
+      - name: Retrieving all database names for this environment
+        id: get_databases
+        # Gets databases filtering by a name. If no filter is provided, it'll
+        # retrieve all databases.
+        run: |
+          # The first time we run api:environments:database-list acli very
+          # kindly also outputs a string asking the user to share
+          # anonymous user data along with the JSON output causing an error
+          # further down the chain - this simply mitigates that.
+          acsf_request=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            --limit='1' \
+            ${{ secrets.ENVIRONMENT_ID }})
+
+          # Gets a list of databases and uses jq to filter the output
+          # to the name value only.
+          database_list=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            ${{ secrets.ENVIRONMENT_ID }} \
+            | jq '.[] | .name')
+
+          # Processes the output of databases to allow the data to
+          # be passed to the 'Backing up databases' step.
+          delimiter="$(openssl rand -hex 8)"
+          echo "db_list<<${delimiter}" >> "${GITHUB_OUTPUT}"
+          echo ${database_list} >> ${GITHUB_OUTPUT}
+          echo "${delimiter}" >> "${GITHUB_OUTPUT}"
+
+      - name: Backing up databases
+        run: |
+
+          # Split db_list into an array.
+          IFS=', ' read -r -a backup_array <<< "${{ steps.get_databases.outputs.db_list }}"
+
+          # Loop through the array of databases and attempt to 
+          # perform a backup for each database.
+          i=0
+          while [ $i -lt ${#backup_array[@]} ]
+          do
```

```suggestion
          while [ $i -lt ${#backup_array[@]} ]; do
```

... though there may be a better way to do this loop, I'm just picking up on this, cos I like having these statements on one line. personal preference 😬 I find it easier to read.